src/Apps/Order/Routes/Shipping2/FulfillmentDetails.tsx

```diff
@@ -341,6 +362,25 @@ const FulfillmentDetailsFormLayout = (props: LayoutProps) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [active, handleSubmit, isValid, values])
 
+  const handleCloseVerification = async () => {
+    await setFieldValue("attributes.addressVerifiedBy", AddressVerifiedBy.USER)
+    await props.addressVerificationComplete()
+  }
+
+  const handleChooseAddress = async (verifiedBy, chosenAddress) => {
+    const newValues = {
+      ...values,
+      attributes: {
+        ...values.attributes,
+        ...chosenAddress,
+        addressVerifiedBy: verifiedBy,
+      },
+    }
+    await setValues(newValues)
+    await props.addressVerificationComplete()
+    submitWithoutFormik(newValues)
+  }
```

Comment:
Side note: It would be great if all of these function callbacks, vs being spread throughout components and passed down as props and calling to parents and so on (which creates a sort of tri-dimensional brittleness) could be rolled up into the context and centralized as actions, or stored within a hook. its not just state stuff that benefits from centralization, but also actions. We have access to formik values and actions from the useFormikContext hook. An action could be plucked from the form context and used directly within the query renderer, no props passing required. 

Basically, the more things that are passed as props, the more complex. The more one centralizes within the new Context and / or leans on hooks, the easier this will be able to change and modify in the future. 

If you're around I can help walk you through this setup; we have an opportunity to centralize 👍 (The best way to think about react in 2023 is: for every passed prop, subtract one point from the game 😄)

Reply:
We talked about this in person and identified a few things to look at. Will update with some changes here.